### PR TITLE
use subtest for table units (pkg/scheduler/api/validation)

### LINKED ISSUE #63267: scheduler: Update tests to use subtest
Go 1.7 added the [subtest feature](https://golang.org/doc/go1.7#testing) which can make table-driven tests much easier to run and debug. Many table-driven tests in `pkg/scheduler` and `test/integration/scheduler` are not using this feature.

As an example, https://github.com/kubernetes/kubernetes/pull/56940 converts some tests to use a sub-benchmark.

Further reading: [Using Subtests and Sub-benchmarks](https://blog.golang.org/subtests)

/kind cleanup
/sig scheduling

### PR DESCRIPTION
**Pull request by xchapter7x**
**What this PR does / why we need it**: Update scheduler's unit table tests to use subtest

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
breaks up PR: https://github.com/kubernetes/kubernetes/pull/63281
/ref #63267

**Release note**:

```release-note
This PR will leverage subtests on the existing table tests for the scheduler units.
Some refactoring of error/status messages and functions to align with new approach.

```

### CODE REVIEW

File: pkg/scheduler/api/validation/validation_test.go [hunk 1]
```diff
@@ -28,44 +28,55 @@ func TestValidatePolicy(t *testing.T) {
...
 			policy:   api.Policy{Priorities: []api.PriorityPolicy{{Name: "NoWeightPriority", Weight: 0}}},
 			expected: errors.New("Priority NoWeightPriority should have a positive weight applied to it or it has overflown"),
 		},
 		{
+			name:     "valid weight priority",
 			policy:   api.Policy{Priorities: []api.PriorityPolicy{{Name: "WeightPriority", Weight: 2}}},
 			expected: nil,
 		},
 		{
+			name:     "invalide negative weight policy",
```
Comment (misterikkit) [inline pkg/scheduler/api/validation/validation_test.go]: typo: invalide

File: pkg/scheduler/api/validation/validation_test.go [hunk 2]
```diff
@@ -28,44 +28,55 @@ func TestValidatePolicy(t *testing.T) {
...
 			policy:   api.Policy{ExtenderConfigs: []api.ExtenderConfig{{URLPrefix: "http://127.0.0.1:8081/extender", PrioritizeVerb: "prioritize", Weight: -2}}},
 			expected: errors.New("Priority for extender http://127.0.0.1:8081/extender should have a positive weight applied to it"),
 		},
 		{
+			name:     "valid filterverb and url prefix",
 			policy:   api.Policy{ExtenderConfigs: []api.ExtenderConfig{{URLPrefix: "http://127.0.0.1:8081/extender", FilterVerb: "filter"}}},
 			expected: nil,
 		},
 		{
+			name:     "valid preemtverb and urlprefix",
```
Comment (misterikkit) [inline pkg/scheduler/api/validation/validation_test.go]: typo: preemtverb

**Review by resouer — APPROVED** [review]
/lgtm

File: pkg/scheduler/api/validation/validation_test.go (hunk 2 above)
Comment (misterikkit) [reply pkg/scheduler/api/validation/validation_test.go]: This typo didn't get fixed, but otherwise LGTM.

Comment (resouer) [reply pkg/scheduler/api/validation/validation_test.go]: Oh, I saw the comment and thought it was after fixed. I'm blind ...

File: pkg/scheduler/api/validation/validation_test.go [hunk 3]
```diff
@@ -28,44 +28,55 @@ func TestValidatePolicy(t *testing.T) {
...
 			policy:   api.Policy{ExtenderConfigs: []api.ExtenderConfig{{URLPrefix: "http://127.0.0.1:8081/extender", PrioritizeVerb: "prioritize", Weight: -2}}},
 			expected: errors.New("Priority for extender http://127.0.0.1:8081/extender should have a positive weight applied to it"),
 		},
 		{
+			name:     "valid filter verb and url prefix",
 			policy:   api.Policy{ExtenderConfigs: []api.ExtenderConfig{{URLPrefix: "http://127.0.0.1:8081/extender", FilterVerb: "filter"}}},
 			expected: nil,
 		},
 		{
+			name:     "valid preemt verb and urlprefix",
```
Comment (misterikkit) [inline pkg/scheduler/api/validation/validation_test.go]: Sorry to keep coming back to typos: s/preemt/preempt/

**Comment by resouer** [comment]
/lgtm

Thanks for fixing.

**Pull request merged**